src/main/java/service/UserService.java

```diff
@@ -0,0 +1,53 @@
+package service;
+
+import db.Database;
+import model.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import util.ObjectMapper;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+
+public class UserService {
+    private Logger logger = LoggerFactory.getLogger(UserService.class);
+
+    /**
+     * 회원가입 메서드
+     * @param parameter 회원가입 폼 입력값. userId, password, name, email
+     * @throws IllegalArgumentException 빈 값이 있을 경우
+     * @throws IllegalStateException 중복된 다른 아이디가 있을 경우
+     */
+    public void join(Map<String, String> parameter) throws IllegalArgumentException, IllegalStateException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        String userId = parameter.get("userId");
+
+        if (Database.findUserById(userId)!=null) {
+            throw new IllegalStateException("중복된 아이디가 있습니다.");
+        }
+
+        User user = (User) new ObjectMapper(User.class, parameter).mapObject();
```

Comment:
```suggestion
        User user = (User) new ObjectMapper(parameter, User.class).mapObject();
```
이게 순서가 더 자연스럽지 않나요? 그냥 제 생각입니다 ㅎㅎ